src/transformers/trainer.py

```diff
@@ -790,6 +793,8 @@ def train(
             model = ShardedDDP(model, self.optimizer)
         elif is_sagemaker_distributed_available():
             model = DDP(model, device_ids=[dist.get_local_rank()], broadcast_buffers=False)
+        if self.deepspeed:
```

Comment:
FYI this breaks most integrations, it should be an `elif` so that we don't fall into the branches after if TPU or sagemaker is here.
Will fix in a commit on master.

Reply:
oh boy, my apologies, my branching skills went haywire yesterday.

Reply:
thank you for the fix, @sgugger

Reply:
No worries, just wanted to alert you :-) Thankfully we found this just before cutting the release candidate!

Reply:
Oh my!

As I said above this literally happened to me several times yesterday, something went haywire and I started adding new branches with just `if`'s adjacent to an existing if/elif/else pile - my brain decided that if they are together it's must be part of the other if/else. So odd. Some new programming language must be percolating through my neurons or a rogue AI took over and is using my brain for its experiments.